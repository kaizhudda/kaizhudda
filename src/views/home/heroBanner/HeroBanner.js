import React from 'react';
import Particles from 'react-particles-js';

import './HeroBanner.scss';

const HeroBanner = () => {
  return (
    <div className="HeroBanner row center-xs">
      <Particles
        className="particles"
        params={{
          "particles": {
            "number": {
              "value": 25
            },
            "size": {
              "value": 3
            }
          },
          "interactivity": {
            "events": {
              "onhover": {
                "enable": true,
                "mode": "repulse"
              }
            }
          }
        }} />
      <div className="content">
        <h1 className="name">Kaiz Hudda</h1>
        <h3 className="title">Front-End Developer</h3>
      </div>
    </div>
  );
}

export default HeroBanner;
