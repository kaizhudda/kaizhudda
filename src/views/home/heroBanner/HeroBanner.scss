@import '../../../styles/variables';
@import '../../../styles/media-queries';

.HeroBanner {
  background: #22a39f;
  color: white;
  font-family: $font-body;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 16px;
  height: 100vh;  
  position: relative;
  min-height: 400px;

  .content {
    position: absolute;
    top: 30%;

    h1.name {
      font-weight: 700;
      font-size: 3.5rem;
      margin-bottom: 10px;
      padding: 25px 100px;
      border: 2px solid rgba(255, 255, 255, 0.35);
  
      @include phone-only {
        padding: 25px 50px;
        font-size: 2rem;
      }
    }
  
    h3.title {
      font-family: $font-code;
      font-weight: 300;
      font-size: 1.75rem;
      margin-top: 15px;
  
      @include phone-only {
        font-size: 1.25rem;
      }
    }
  }


}