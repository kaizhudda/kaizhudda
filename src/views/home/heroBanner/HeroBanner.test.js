import React from 'react';
import { shallow } from 'enzyme';
import HeroBanner from './HeroBanner';

let wrapped = shallow(<HeroBanner />); 

describe('HeroBanner', () => {
  it('should render the HeroBanner Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should contain a name', () => {
    expect(wrapped.find('.name').text()).toEqual('Kaiz Hudda');
  });

  it('should contain a role', () => {
    expect(wrapped.find('.title').text()).toEqual('Front-End Developer');
  });
});
