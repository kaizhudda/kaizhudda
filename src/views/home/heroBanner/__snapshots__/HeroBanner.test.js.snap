// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner should render the HeroBanner Component correctly 1`] = `
<div
  className="HeroBanner row center-xs"
>
  <e
    className="particles"
    height="100%"
    params={
      Object {
        "interactivity": Object {
          "events": Object {
            "onhover": Object {
              "enable": true,
              "mode": "repulse",
            },
          },
        },
        "particles": Object {
          "number": Object {
            "value": 25,
          },
          "size": Object {
            "value": 3,
          },
        },
      }
    }
    style={Object {}}
    width="100%"
  />
  <div
    className="content"
  >
    <h1
      className="name"
    >
      Kaiz Hudda
    </h1>
    <h3
      className="title"
    >
      Front-End Developer
    </h3>
  </div>
</div>
`;
